File: AutomatonAlgorithm/src/Automaton.java
/**
 * Created by jenamorado on 7/25/16.
 */
import java.util.ArrayList;

/**
 * Created by jenamorado on 7/25/16.
 */
public class Automaton {
    ArrayList<States> states;
    ArrayList<Character> alphabet;
    String strToEval;

    public States getInitialState(){
        for(int j = 0; j < states.size(); j++){
            if(states.get(j).isAcceptanceState){
                return states.get(j);
            }
        }
        return null;
    }

    public States getState(String stateName){
        for(int j = 0; j < states.size(); j++){
            if(states.get(j).getName().equals(stateName)){
                return states.get(j);
            }
        }
        return null;
    }

    public boolean evaluate(String strEvString){
        char eval[] = strEvString.toCharArray();
        States currentState = getInitialState();
        if(currentState!=null){
            for(int i = 0; i < eval.length; i++){
                for(int j = 0; j < currentState.transitions.size(); j++){
                    if(currentState.transitions.get(j).symbol == eval[i]){
                        currentState = getState(currentState.transitions.get(j).destiny);
                    }
                }
            }
            if(currentState.isAcceptanceState){
                return true;
            }else {
                return false;
            }
        }
        return false;
    }
}

File: AutomatonDfa/src/Automaton.kt
import java.util.ArrayList

/**
 * Created by jenamorado on 7/25/16.
 */
class Automaton {
    open var states: ArrayList<States>? = null
    open var alphabet: ArrayList<Char>? = null
    open var strToEval: String? = null

    val initialState: States?
        get() {
            for (j in states!!.indices) {
                if (states!![j].isAcceptanceState) {
                    return states!![j]
                }
            }
            return null
        }

    fun getState(stateName: String?): States? {
        for (j in states!!.indices) {
            if (states!![j].name == stateName) {
                return states!![j]
            }
        }
        return null
    }

    fun evaluate(strEvString: String): Boolean {
        val eval = strEvString.toCharArray()
        var currentState = initialState
        if (currentState != null) {
            for (i in eval.indices) {
                for (j in currentState!!.transitions!!.indices) {
                    if (currentState.transitions!![j].symbol == eval[i]) {
                        currentState = getState(currentState.transitions!![j].destiny)
                    }
                }
            }
            if (currentState!!.isAcceptanceState) {
                return true
            } else {
                return false
            }
        }
        return false
    }
}
